x-pack/plugins/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.ts

```diff
@@ -44,6 +45,14 @@ import { EvaluatedRuleParams, evaluateRule } from './lib/evaluate_rule';
 import { MissingGroupsRecord } from './lib/check_missing_group';
 import { convertStringsToMissingGroupsRecord } from './lib/convert_strings_to_missing_groups_record';
 
+export type MetricThresholdAlert = Omit<
+  ObservabilityMetricsAlert,
+  'kibana.alert.evaluation.values'
+> & {
+  // Defining a custom type for this because the schema generation script doesn't allow explicit null values
+  'kibana.alert.evaluation.values'?: Array<number | null>;
+};
```

Comment:
We auto-generate a schema for each context based on the registered field map. The auto-generated schema can be found at `packages/kbn-alerts-as-data-utils/src/schemas/generated/observability_metrics_schema.ts`. However, the mapping definition for `kibana.alert.evaluation.values` is an array of `scaled_float` values which translates to a schema `Array<string | number>` (since ES accepts numerical strings and will coerce into number). However, the type that's generated by the code is `Array<number | null>`. To accommodate, I'm modifying the auto-generated schema (which is just meant to be a suggestion based on the field map anyway).